### PR TITLE
Document process in failed main builds

### PR DESCRIPTION
**Pull request by jhchabran**
See #102

### CODE REVIEW

File: content/engineering/continuous_releasability.md [hunk 1]
```diff
@@ -44,6 +41,27 @@ Feature flags achieve 2 things:
   Releasability Contract. E.g., if we discover a critical bug in a new feature 1 day before release,
   we can immediately disable it to make the release deadline.
 
+### A red `main` build is not okay and must be fixed
+
+> In essence: Someone must have eyes on the build failure. Unsure about what's happening? Get help on #buildkite-main.
+
+- When a PR breaks the build, the author is responsible for investigating why, and asking for help if necessary:
+  - The failure will appear on [#buildkite-main](https://sourcegraph.slack.com/archives/C02FLQDD3TQ).
+  - If after half an hour the cause is still unclear, ask for help!
```
Comment (pdubroy) [inline content/engineering/continuous_releasability.md]:
```suggestion
  - If you've done ~30 mins of investigation and the cause is still unclear, ask for help!
```

File: content/engineering/continuous_releasability.md [hunk 2]
```diff
@@ -44,6 +41,27 @@ Feature flags achieve 2 things:
...
+  - Handing the issue over to someone else (for any reason) is totally okay, but it has to happen.
+  - If there's no action being taken after a reasonable amount of time, the offending PR can be reverted by anyone blocked by it.
+- If there is reasonable suspicion of a flake (e.g. can't reproduce the problem locally) or if it’s clear that the cause is not related to the PR:
+  - Rebuild the job.
+  - Notify the team in charge of the concerned test.
+  - It's a CI flake? Pass ownership to the DX team.
+- If there is no immediate fix in sight (or rebuilding didn't fix it):
+  - Mark the faulty test ask skipped or revert the changes to restore the main branch to green and avoid blocking others.
```
Comment (pdubroy) [inline content/engineering/continuous_releasability.md]:
```suggestion
  - Mark the faulty test as skipped or revert the changes to restore the main branch to green and avoid blocking others.
```

**Comment by jhchabran** [comment]
Preview: https://deploy-preview-236--sourcegraph-handbook.netlify.app/engineering/continuous_releasability#a-red-main-build-is-not-okay-and-must-be-fixed 

(@felixfbecker ☝️ this is awesome)

File: content/engineering/continuous_releasability.md [hunk 3]
```diff
@@ -44,6 +41,27 @@ Feature flags achieve 2 things:
...
+  - if reverting won't fix because it depends on external resources, just comment out that test and open a ticket mentioning the owners.
```
Comment (pdubroy) [inline content/engineering/continuous_releasability.md]: I'm not quite sure what this one means.

**Review by pdubroy — APPROVED** [review]

**Comment by bobheadxi** [comment]
I actually feel like all this should go in docs.sourcegraph.com instead, since it pertains to Sourcegraph development as a whole and not Sourcegraph as a company 🤔 We already have bits and pieces in both places, so we might want to unify them as well

File: content/engineering/continuous_releasability.md (hunk 3 above)
Comment (jhchabran) [reply content/engineering/continuous_releasability.md]:
```suggestion
  - In some edge cases, reverting a PR won't solve the problem (because it depends on an external resource that is in an incorrect state), mark that test as skipped and open a ticket mentioning the owners.
```

@pdubroy Is it clearer now? My bad for pushing such incorrectly phrased sentence, I forgot to double check that the text from the PM was correct enough.

**Comment by jhchabran** [comment]
> I actually feel like all this should go in docs.sourcegraph.com instead, since it pertains to Sourcegraph development as a whole and not Sourcegraph as a company 🤔 We already have bits and pieces in both places, so we might want to unify them as well

That's a good question. If we were to move this in the docs, we should at least have a link toward it here though, as the process is very relevant to that page. 

@pdubroy what's your take on this? 

1. Move it here https://github.com/sourcegraph/sourcegraph/tree/main/doc/dev
2. Keep it here 
3. It's blurry and there is no hard rule 😄

**Comment by bobheadxi** [comment]
+1 to move to doc/dev and link from here

**Comment by pdubroy** [comment]
Yeah, feels like it belongs under doc/dev

File: content/engineering/continuous_releasability.md (hunk 3 above)
Comment (pdubroy) [reply content/engineering/continuous_releasability.md]: Sounds good, thanks!

**Comment by jhchabran** [comment]
Alright, moving it over there then :)
